File: core-platform/apps/core-webapp/src/services/performanceReview.service.ts
import axiosInstance from '../axiosInstance';
import type { PerformanceReviewDTO } from '../types/performance.types';

const PERFORMANCE_REVIEWS_API_BASE = '/performance/reviews';

export const performanceReviewService = {
  // Submit a review for a request
  submitReview: async (requestId: number, data: PerformanceReviewDTO): Promise<PerformanceReviewDTO> => {
    const response = await axiosInstance.post(`${PERFORMANCE_REVIEWS_API_BASE}/${requestId}/submit`, data);
    return response.data;
  },

  // Get all reviews for an employee
  getEmployeeReviews: async (employeeId: number): Promise<PerformanceReviewDTO[]> => {
    const response = await axiosInstance.get(`${PERFORMANCE_REVIEWS_API_BASE}/employee/${employeeId}`);
    return response.data;
  },

  // Get all reviews for a cycle
  getCycleReviews: async (cycleId: number): Promise<PerformanceReviewDTO[]> => {
    const response = await axiosInstance.get(`${PERFORMANCE_REVIEWS_API_BASE}/cycle/${cycleId}`);
    return response.data;
  },
};

File: core-platform/services/core-service/src/main/java/com/dev/core/service/performance/PerformanceReviewService.java
package com.dev.core.service.performance;

import java.util.List;
import com.dev.core.model.performance.PerformanceReviewDTO;

public interface PerformanceReviewService {

    PerformanceReviewDTO submitReview(Long requestId, PerformanceReviewDTO dto);

    List<PerformanceReviewDTO> getReviewsForEmployee(Long employeeId);

    List<PerformanceReviewDTO> getReviewsForCycle(Long cycleId);

}
